Assingment7-scf/SparseMatrix.java

```diff
@@ -0,0 +1,176 @@
+	 * @param sparseObj_2 second matrix to be added.
+	 * @throws AssertionError when dimensions are different.
+	 * @return the added matrix.
+	 */
+	public static int[][] addMatrix(SparseMatrix sparseObj_1, SparseMatrix sparseObj_2) throws AssertionError{
+		if(sparseObj_1.totalCol != sparseObj_2.totalCol || sparseObj_1.totalRow != sparseObj_2.totalRow)
+			throw new AssertionError("Dimensions are different");
+
+		int[][] newMatrix = new int[sparseObj_1.totalRow][sparseObj_1.totalCol]; 
+
+		for(int i=0; i<sparseObj_1.matrixLoc.length;i++){
+			newMatrix[sparseObj_1.matrixLoc[i].row][sparseObj_1.matrixLoc[i].col] = sparseObj_1.matrixLoc[i].value; 
+		}
+
+		for(int i=0; i<sparseObj_2.matrixLoc.length;i++){
+			newMatrix[sparseObj_2.matrixLoc[i].row][sparseObj_2.matrixLoc[i].col] += sparseObj_2.matrixLoc[i].value; 
+		}
+		return newMatrix;
+	}
+
+	/**
+	 * This function multiplies two matrices.
+	 * @param sparseObj_1 First matrix to multiply.
+	 * @param sparseObj_2 second matrix to multiply.
+	 * @return the added matrix.
+	 * @throws AssertionError when obj.col != obj.row;
+	 */
+	public static int[][] multiplyMatrix(SparseMatrix sparseObj_1, SparseMatrix sparseObj_2) throws AssertionError{
+		if(sparseObj_1.totalCol != sparseObj_2.totalRow)
+			throw new AssertionError("Mismatch in dimensions of col*row of matrices.");
+
+		int[][] newMatrix = new int[sparseObj_1.totalRow][sparseObj_2.totalCol];
+
+		for(int i =0 ;i<sparseObj_1.matrixLoc.length;i++){
+			for(int j=0;j<sparseObj_2.matrixLoc.length;j++){
+				if(sparseObj_1.matrixLoc[i].col == sparseObj_2.matrixLoc[j].row)
+					newMatrix[sparseObj_1.matrixLoc[i].row][sparseObj_2.matrixLoc[j].col] += sparseObj_1.matrixLoc[i].value * sparseObj_2.matrixLoc[j].value;
+			}
+		}
+		return newMatrix;
+	}
+
+	/*	public void print(){
+		for(int i=0;i<matrixLoc.length;i++){
+			System.out.println("obj : " + i);
+			System.out.println("row : "+matrixLoc[i].row + " col : " + matrixLoc[i].col + " value : "+matrixLoc[i].value);
+		}
+	}
+
+	public static void main(String[] args){
+		int[][] arr = 
+				{{5,0,3},
+				{2,0,0},
+				{1,0,0}};
+		SparseMatrix s1 = new SparseMatrix(arr);
+		int[][] b = s1.transposeOfMatrix();
+
+		for(int i=0;i<b.length;i++){
+			for(int j=0; j<b[0].length;j++){
+				System.out.print(" " + b[i][j] + " ");
+			}
+			System.out.println();
+		}
+
+		int[][] brr = 
+			{{0,0,0,1},
+			{2,3,0,0},
+			{1,0,1,0}};
+		SparseMatrix s2 = new SparseMatrix(brr);
+
+		boolean check = s1.checkSymmetric();
+		System.out.println(check);
+
+		b = s1.addMatrix(s1, s2);
+		for(int i=0;i<b.length;i++){
+			for(int j=0; j<b[0].length;j++){
+				System.out.print(" " + b[i][j] + " ");
+			}
+			System.out.println();
+		}
+
+		System.out.println("---------multiply----------");
+		b = s1.multiplyMatrix(s1, s2);
+		for(int i=0;i<b.length;i++){
+			for(int j=0; j<b[0].length;j++){
+				System.out.print(" " + b[i][j] + " ");
+			}
+			System.out.println();
+		}
+	}*/
+}
```

Comment:
a lot of commented code

Reply:
This commented part is for main() method, as I was trying the code using main() method before junit testing. So after creating Testing class I've commented the main() rather than deleting it.

Reply:
Either remove it or uncomment it, we do not allow commented code in projects, so get used to it.